helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml

```diff
@@ -146,10 +173,45 @@ spec:
           items:
             - key: relay
               path: relay.yaml
+      {{- if eq (include "splunk-otel-collector.distribution" .) "eks/fargate" }}
+      - name: {{ template "splunk-otel-collector.eksFargateClusterReceiverScript" . }}
+        configMap:
+          name: {{ template "splunk-otel-collector.eksFargateClusterReceiverScript" . }}
+          items:
+            - key: script
+              path: eks-fargate-otelcol-with-env.sh
+              mode: 0777
+      {{- end }}
+      {{- if eq (include "splunk-otel-collector.clusterReceiverNodeDiscovererInitContainerEnabled" .) "true" }}
+      - name: {{ template "splunk-otel-collector.clusterReceiverNodeDiscovererScript" . }}
+        configMap:
+          name: {{ template "splunk-otel-collector.clusterReceiverNodeDiscovererScript" . }}
+          items:
+            - key: script
+              path: lookup-eks-fargate-receiver-node.sh
+              mode: 0777
+      - name: messages
+        emptyDir: {}
+      {{- end }}
       {{- if $clusterReceiver.extraVolumes }}
       {{- toYaml $clusterReceiver.extraVolumes | nindent 6 }}
       {{- end }}
       {{- if $clusterReceiver.priorityClassName }}
       priorityClassName: {{ $clusterReceiver.priorityClassName }}
       {{- end }}
 {{- end }}
+
+{{- if eq (include "splunk-otel-collector.distribution" .) "eks/fargate" }}
+---
+apiVersion: v1
+kind: Service
```

Comment:
Why service is added?

Reply:
```StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service.```

- https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

Reply:
Without a headless service and link:

```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec
error running install splunk-otel-collector-chart: exit status 1
```